src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/page/RichTextEditorTextArea.kt

```diff
@@ -0,0 +1,59 @@
+package com.atlassian.performance.tools.jiraactions.api.page
+
+import com.atlassian.performance.tools.jiraactions.Patience
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+import org.openqa.selenium.By
+import org.openqa.selenium.JavascriptExecutor
+import org.openqa.selenium.WebDriver
+import org.openqa.selenium.WebElement
+import org.openqa.selenium.support.ui.ExpectedConditions
+import org.openqa.selenium.support.ui.ExpectedConditions.*
+import java.time.Duration
+
+/**
+ * Helper class for Rich Text Editor enabled Textarea
+ *
+ */
+class RichTextEditorTextArea(
+    private val driver: WebDriver
+) {
+
+    private val logger: Logger = LogManager.getLogger(this::class.java)
+
+    private fun clearAndTypeIfPresent(id :String, text: String) {
+        if (driver.isElementPresent(By.id(id))) {
+            val webElement = driver.findElement(By.id(id))
+            driver.wait(timeout = Duration.ofSeconds(2), condition = elementToBeClickable(webElement))
+            webElement.clear()
+            webElement.sendKeys(text)
+        }
+    }
+
+    fun clearAndTypeIfPresentRte(id : String, text: String) {
+        //check richeditor-cover exists in class attribute
+        if (driver.isElementPresent(By.id(id))) {
+            val webElement = driver.findElement(By.id(id))
+            val attrClass = webElement.getAttribute("class")
+            if(attrClass != null && attrClass.contains(other = "richeditor-cover", ignoreCase = true)){
+                logger.debug("richeditor detected, will try Rich Text Editer...")
+                val iframeXpath = "//div[textarea[@id='$id']]//iframe"
+                driver.wait(timeout = Duration.ofSeconds(10), condition = frameToBeAvailableAndSwitchToIt(By.xpath(iframeXpath)))
+
+                val bodyElement = driver.findElement(By.id("tinymce"))
+                //data-projectkey={project}
+                driver.wait(timeout = Duration.ofSeconds(10), condition = ExpectedConditions.attributeToBeNotEmpty(bodyElement, "data-projectkey"))
+                bodyElement.clear()
+                bodyElement.sendKeys(text)
+
+                driver.switchTo().parentFrame()
+            }
+            else{
+                //try plain text editing
+                logger.debug("Try use Raw Text Editer...")
+                clearAndTypeIfPresent(id, text)
```

Comment:
This `if`/`else` looks a bit imbalanced.
The `if` has 7 logical lines and works on low-level WebDriver logic.
The `else` has 1 logical line (+1 debugging line) and operates on a high-level page object logic.

I'd pack up the `if` contents into a `private fun` to see basically say:
```kotlin
if (isRte(element)) {
    clearAndTypeRichTextIfPresent(element, text)
} else {
    clearAndTypePlainTextIfPresent(element, text)
}
```

Or if we apply the other suggestions:
```kotlin
if (isRich(element)) {
    overwriteRich(element, text)
} else {
    overwritePlain(element, text)
}
```